assets/js/admin/marketplace-suggestions.js

```diff
@@ -0,0 +1,340 @@
+			} );
+
+			// hide promos the user has dismissed
+			promos = _.filter( promos, function( promo ) {
+				return ! _.contains( marketplace_suggestions.dismissed_suggestions, promo.slug );
+			} );
+
+			// hide promos for things the user already has installed
+			promos = _.filter( promos, function( promo ) {
+				return ! _.contains( marketplace_suggestions.installed_woo_plugins, promo['hide-if-installed'] );
+			} );
+
+			// hide promos that are not applicable based on user's installed extensions
+			promos = _.filter( promos, function( promo ) {
+				if ( ! promo['show-if-installed'] ) {
+					// this promotion is relevant to all
+					return true;
+				}
+
+				// if the user has any of the prerequisites, show the promo
+				return ( _.intersection( marketplace_suggestions.installed_woo_plugins, promo['show-if-installed'] ).length > 0 );
+			} );
+
+			return promos;
+		}
+
+		function hidePageElementsForEmptyState( visibleSuggestions ) {
+			var showingProductsEmptyStateSuggestions = _.contains( visibleSuggestions, 'products-list-empty-body' );
+			if ( showingProductsEmptyStateSuggestions ) {
+				$( '#screen-meta-links' ).hide();
+				$( '#wpfooter' ).hide();
+			}
+			if ( ! showingProductsEmptyStateSuggestions ) {
+				// hide the header & footer, they don't make sense without specific promotion content
+				$( '.marketplace-suggestions-container[data-marketplace-suggestions-context="products-list-empty-header"]' ).hide();
+				$( '.marketplace-suggestions-container[data-marketplace-suggestions-context="products-list-empty-footer"]' ).hide();
+			}
+		}
+
+		function displaySuggestions( marketplaceSuggestionsApiData ) {
+			var visibleSuggestions = [];
+
+			// iterate over all suggestions containers, rendering promos
+			$( '.marketplace-suggestions-container' ).each( function() {
+				// determine the context / placement we're populating
+				var context = this.dataset.marketplaceSuggestionsContext;
+
+				// find promotions that target this context
+				var promos = getRelevantPromotions( marketplaceSuggestionsApiData, context );
+
+				// render the promo content
+				for ( var i in promos ) {
+					var linkText = promos[ i ]['link-text'];
+					var linkoutIsButton = false;
+					if ( promos[ i ]['button-text'] ) {
+						linkText = promos[ i ]['button-text'];
+						linkoutIsButton = true;
+					}
+
+					// dismiss is allowed by default
+					var allowDismiss = true;
+					if ( promos[ 0 ]['allow-dismiss'] === false ) {
+						allowDismiss = false;
+					}
+
+					var content = renderListItem(
+						promos[ i ].slug,
+						promos[ i ].icon,
+						promos[ i ].title,
+						promos[ i ].copy,
+						promos[ i ].url,
+						linkText,
+						linkoutIsButton,
+						allowDismiss
+					);
+					$( this ).append( content );
+					$( this ).addClass( 'showing-suggestion' );
+					visibleSuggestions.push( promos[i].context );
+
+					window.wcSettings.recordEvent( 'marketplace_suggestion_displayed', {
```

Comment:
Need to check if this is permitted on .org. Tracks is opt-in but the rule is "Third-party advertisement mechanisms which track usage and/or views." so I don't know if that is an exclusion.

Reply:
We're getting this clarified, it's a good point. Will likely follow up separate to this PR.